CMakeLists.txt

```diff
@@ -1478,18 +1484,26 @@ else()
     hpx_add_compile_flag_if_available(-wd2536)
   endif()
 
+
   # rdtsc is an x86 instruction that reads the value of a CPU time stamp
   # counter. rdtscp is an extension to rdtsc. The difference is that rdtscp is
   # a serializing instruction.
   hpx_cpuid("rdtsc" HPX_WITH_RDTSC
     DEFINITIONS HPX_HAVE_RDTSC)
 
+ # One can not assume if RDTSCP is available on the hardware
+ # of the build infrastructure, that it will be available on
+ # all potential target hardware, see Issue  #3575
+ if(NOT $HPX_WITH_BUILD_BINARY_PACKAGE)
```

Comment:
`HPX_WITH_BUILD_BINARY_PACKAGE` needs to be wrapped in curly braces. Could you also clean up the history? Thanks!

Reply:
Added